subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -335,19 +325,7 @@ static bool schedule_send(struct bt_mesh_ext_adv *ext_adv)
 	}
 
 	atomic_clear_bit(ext_adv->flags, ADV_FLAG_SCHEDULE_PENDING);
-
-	if ((IS_ENABLED(CONFIG_BT_MESH_ADV_EXT_FRIEND_SEPARATE) &&
-	     ext_adv->tags & BT_MESH_ADV_TAG_BIT_FRIEND) ||
-	    (CONFIG_BT_MESH_RELAY_ADV_SETS > 0 && ext_adv->tags & BT_MESH_ADV_TAG_BIT_RELAY)) {
-		k_work_reschedule(&ext_adv->work, K_NO_WAIT);
-	} else {
-		/* The controller will send the next advertisement immediately.
-		 * Introduce a delay here to avoid sending the next mesh packet closer
-		 * to the previous packet than what's permitted by the specification.
-		 */
-		delta = k_uptime_delta(&timestamp);
-		k_work_reschedule(&ext_adv->work, K_MSEC(ADV_INT_FAST_MS - delta));
-	}
+	k_work_submit(&ext_adv->work);
```

Comment:
Putting comment here as I can't put it at line 266. `schedule_send` at line 266 should be redesigned as it will just run the same `send_pending_adv` work again.

Reply:
No, it shouldn't. The same work will not be run since flag `ADV_FLAG_ACTIVE` has been already cleared after sending. Even more, if work is submitted again while the ongoing work was in the submitted queue, this work will be lost. Probably adv structure will leak over fifo since it is put to fifo but task to get it is not submitted.
Without this, about 30-40% bsim tests do not work.

Reply:
By redesign I don't mean you need to not call `schedule_send` in other places. But you can go through these flags checks and if it comes to `k_work_submit` just continue here.

Reply:
Something like this: https://github.com/PavelVPV/zephyr/commit/cd923d784ff4a6ea854326fc8df1aa27e3a3286d

Reply:
I'm sure it is not worth to mix the redesign of internal states aka flags handling (very ugly approach comparing to normal fsm) with removing delay task. They are not related to each other. If we do not want replanning work over work, it should be done separately and shouldn't block ongoing changes. 
Removing of delay doesn't change work usage paradigma.

Reply:
Please see the commit from the message above. And yes, perhaps flags can also be redesigned.

Reply:
I do not think I agree with suggested changes, since they really submit work twice.
As I said before, refactoring should be done as separate task with fixing of impacted bsim tests.

Reply:
> I do not think I agree with suggested changes, since they really submit work twice.

As you may see I removed re-submitting the work from `send_pending_adv`.

> As I said before, refactoring should be done as separate task with fixing of impacted bsim tests.

They shouldn't be impacted really if the behavior of the advertiser stays the same. Only the blob transfer in push mode failed at my desk with the commit above when I ran it in parallel mode, but didn't fail when I ran it separately which is suspicious. If you feel OK with this change, then fine, I'll create a task to fix this later.

Reply:
NCSDK-25117